queue.js

```diff
@@ -288,19 +288,19 @@ function queueAsPromised (context, worker, _concurrency) {
   }
 
   function drained () {
-    if (queue.idle()) {
-      return new Promise(function (resolve) {
-        resolve()
-      })
-    }
-
-    var previousDrain = queue.drain
-
     var p = new Promise(function (resolve) {
-      queue.drain = function () {
-        previousDrain()
-        resolve()
-      }
+      setImmediate(() => {
```

Comment:
Could you check if a queueMicroTask() call does the job as well? I would prefer not to turn the event loop.

Reply:
> Could you check if a queueMicroTask() call does the job as well? I would prefer not to turn the event loop.

That won't run on older versions of node though and I noticed in other PRs that you want it to be compatible all the way back essentially.

process.nextTick() does the job and is even higher priority than queueMicroTask(), so I can use that if you like? My worry was adding the ability to potentially starve the event loop, although it shouldn't be possible with normal usage.

Let me know how you'd like to proceed.

Reply:
`nextTick` is perfect here, thanks!